build/kokoro/build.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+cd "$(dirname "$0")"
+
+
+##
+#
+# Environment Variables (required):
+#
+# JOB_MODE: 'integration' for stable or 'integration-dev' for adhoc experimental
+# LANGUAGE: 'java' | 'go' | 'cpp' | 'interop'
+#
+##
+
+PROJECT=microsvcs-testing
+CLUSTER=grpc-o11y-integration-testing-cluster
+ZONE=us-central1-c
+RELEASE=1.53.0-dev
+TEST_DIR=`realpath ../../observability/test`
+
+if [ -z "${JOB_MODE}" ] || [ -z "${LANGUAGE}" ] ; then
+  echo "Error: env var JOB_MODE and LANGUAGE are required."
+  exit 1
+fi
+
+if [ -z "${KOKORO_ARTIFACTS_DIR}" ] ; then
+  echo "Error: env var KOKORO_ARTIFACTS_DIR is not set."
+  exit 1
+fi
+
+
+
+##
+#
+# Install pre-requisites and set up environment
+#
+##
+
+gcloud config set project ${PROJECT}
+
+sudo apt-get -qq update
+sudo apt-get -qq install -y \
+  google-cloud-sdk-gke-gcloud-auth-plugin \
+  kubectl clang
+
+# Install Python packages
+python3 --version
+python3 -m pip install -r requirements.txt
```

Comment:
I would recommend splitting this into two files:
 - `requirements.txt` with version ranges specified, but not specific versions
 - `requirements.lock` generated from `requirements.txt` with specific versions specified. These are the versions actually used in test runs.

We have seen several PSM interop outages caused by pushes of new versions of Python packages. By freezing to particular packages, you'll protect yourself from non-hermetic breakage (and a lot of headaches).

Reply:
This has been implemented. Added a `requirements.lock` file generated from running `pip freeze` in a virtualenv.